subsystems/drivebase/SwerveDrive.java

```diff
@@ -311,8 +327,24 @@ public void mustangPeriodic() {
         Logger.recordOutput(DRIVEBASE_PITCH, getPitch());
         Logger.recordOutput(DRIVEBASE_ROLL, getRoll());
 
+       
+
+        if (kConfig.kDriveMotorType.equals(Motor_Type.KRAKEN_X60)) {
+            emaCurrent = currentAlpha * ((TalonFX) mModules[0].getDriveMotor()).getSupplyCurrent() + (1 - currentAlpha) * emaCurrent; // weighted current average of front-right drive motor
+            if (emaCurrent > 30) { // TODO tune this? what should be the threshold?
+                currentLimitConfigs.SupplyCurrentLimit = currentSupplyCurrentLimit - 0.25; // TODO what increment to decrease/increase limit? how frequently current readings update?
+            } else {
+                currentLimitConfigs.SupplyCurrentLimit = currentSupplyCurrentLimit + 0.25;
+            }
+            currentLimitConfigs.SupplyCurrentThreshold = currentLimitConfigs.SupplyCurrentLimit + 5; // if the current goes at or above supply current threshold for more than 0.5 sec (time threshold), limited to supply current limit
+            driveMotorConfig.withCurrentLimits(currentLimitConfigs);
+    
+            ((TalonFX) mModules[0].getDriveMotor()).getConfigurator().apply(driveMotorConfig); // apply new current limit to all, assuming they're correlated
```

Comment:
there ought to be a case (the by far most common case) where we do nothing here because there's no need to change anything